numpy/lib/function_base.py

```diff
@@ -2669,6 +2689,18 @@ class vectorize(object):
            <http://docs.scipy.org/doc/numpy/reference/c-api.generalized-ufuncs.html>`_.
     """
 
+    def __new__(cls, pyfunc=None, *args, **kw):
+        def new_with_kw(pyfunc):
```

Comment:
This should be called something that is useful to an end user inspecting the unapplied decorator, like `vectorize_decorator`